fvm/environment/programs.go

```diff
@@ -127,6 +138,47 @@ func (programs *Programs) get(
 	return nil, false
 }
 
+// popDependencyTracker removes the dependencyTracker for the given location,
+// and returns the list of dependencies.
+func (programs *Programs) popDependencyTracker(
+	location common.AddressLocation,
+) derived.ProgramDependencies {
+	dependencies, ok := programs.dependencyTracker[location]
+	if !ok {
+		return make(derived.ProgramDependencies)
```

Comment:
I have concerns with cadence calling Set without calling Get first. There seems to be a couple of instances of that. I 'll verify they don't mess this up and change this to a stack.

Reply:
txnState BeginParseRestrictedNestedTransaction / CommitParseRestricted already guard against this.  if commit and called before begin, this would error out